### PR TITLE
Speedup librispeech dataset processing by using multiple processes.

### PR DESCRIPTION
**Pull request by mousebaiker**
It is 2021, we should use parallel computations for slow scripts when possible.

As each directory contains independent data, we can easily add more processes to get_librispeech_data script to decrease waiting time.

On my machine the speedup when using 4 processes was 7-8 fold.
The whole download process went from >2h waiting time to 15 minutes.

The downloaded data is identical, so there should not be any problems introduced by parallelizing the script.

### CODE REVIEW

File: scripts/dataset_processing/get_librispeech_data.py [hunk 1]
```diff
@@ -93,27 +129,13 @@ def __process_data(data_folder: str, dst_folder: str, manifest_file: str):
...
-                entry = {}
-                entry['audio_filepath'] = os.path.abspath(wav_file)
-                entry['duration'] = float(duration)
-                entry['text'] = transcript_text
-                entries.append(entry)
+            files.append(os.path.join(root, filename))
+
+    with multiprocessing.Pool(num_workers) as p:
+        processing_func = functools.partial(__process_transcript, dst_folder=dst_folder)
+        results = p.imap_unordered(processing_func, files)
```
Comment (titu1994) [inline scripts/dataset_processing/get_librispeech_data.py:136]: This is unordered map - changes order of validation data files which will make it hard to process data (equivalent to shuffling validation and text data). Use joblib.Parallel or mp.map() for ordered results.

**Review by titu1994 — CHANGES_REQUESTED** [review]

File: scripts/dataset_processing/get_librispeech_data.py (hunk 1 above)
Comment (mousebaiker) [reply scripts/dataset_processing/get_librispeech_data.py]: Thanks for the catch about the order.

I decided to got with imap. It guarantees order and I checked on my machine that the order is preserved. 
Compared to joblib, it is much easier to couple tqdm with it. And having progress bar is very informative for these long scripts.

**Comment by mousebaiker** [comment]
PTAL. I addressed the ordering issue.

**Review by titu1994 — APPROVED** [review]

**Comment by titu1994** [comment]
LGTM will merge after tests pass

**Pull request merged**